File: Week 1 - Reverse Words in Paragraph/Multi_Threading_Reverse_String_In_Paragraph.py
import threading

def reverseWord(word,lst):
    return lst.append(word[::-1])

para = "Hi there! My name is Ujjwal"

paraList = para.split(' ')
reversedParaList = []

threads = []
for word in paraList:
    thread = threading.Thread(target=reverseWord, args=(word,reversedParaList))
    threads.append(thread)

for thread in threads:
    thread.start()
    
for thread in threads:
    thread.join()

print(*reversedParaList)

File: Week 1 - Reverse Words in Paragraph/Multi_Threading_Reverse_String_In_Paragraph.java
import java.util.*;

public class reversestring extends Thread {
    private String para;

    public reversestring(String para) {
        this.para = para;
    }

    @Override
    public void run() {
        String[] words = para.split("\\s");
        StringBuilder sb = new StringBuilder();
        for (String word : words) {
            sb.append(new StringBuilder(word).reverse().toString() + " ");
        }
        para = sb.toString().trim();
    }

    public String getPara() {
        return para;
    }

    public static void main(String[] args) {
        Scanner s= new Scanner(System.in);
        String str= s.nextLine();   
        reversestring thread = new reversestring(str);
        thread.start();
        thread.join()
        String reversedSentence = thread.getPara();
        System.out.println(reversedSentence);
    }
}
